Sources/Jobs/Scheduler/DateExtensions/DateExtensionComparison.swift

```diff
@@ -0,0 +1,52 @@
+import Foundation
+
+extension Date {
+    /// Returns the amount of years from another date
+    func years(from date: Date) -> Int? {
+        return Calendar.current.dateComponents([.year], from: date, to: self).year ?? nil
+    }
+
+    func quarters(from date: Date) -> Int? {
+        // bug in quarter on ios12
+        //return Calendar.current.dateComponents([.quarter], from: date, to: self).quarter ?? nil
```

Comment:
Have we checked if this works on Linux? Cause we're not really concerned if there's a bug on ios12